### PR TITLE
Fix encoding for Elasticsearch count pushdown

### PR DESCRIPTION
**Pull request by bvolpato**
## Description

We've identified that when a COUNT(*) query was pushed down and contained special characters, the QueryBuilder string was being handled as ISO-8859-1 and causing parsing issues for Elasticsearch.

## Additional context and related issues

For example, this query:
```
SELECT COUNT(*) FROM catalog.default.users where country = 'Türkiye';
```

In case the "country" field is a keyword, would result in:

> {"error":{"root_cause":[{"type":"parsing_exception","reason":"Failed to parse","line":1,"col":53}],"type":"parsing_exception","reason":"Failed to parse","line":1,"col":53,"caused_by":{"type":"x_content_parse_exception","reason":"[1:53] [bool] failed to parse field [filter]","caused_by":{"type":"json_parse_exception","reason":"Invalid UTF-8 middle byte 0x73\n at [Source: (org.elasticsearch.common.io.stream.ByteBufferStreamInput); line: 1, column: 64]"}}},"status":400}

The source for the problem was the `new StringEntity(sourceBuilder.toString())`, which uses https://github.com/apache/httpcomponents-core/blob/rel/v4.4.16/httpcore/src/main/java/org/apache/http/entity/ContentType.java#L106-L107 and defaults to ISO.

image

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 1]
```diff
@@ -1536,6 +1536,37 @@ public void testFilters()
...
+        String indexName = "filter_charset_pushdown";
+
+        @Language("JSON")
+        String mappings = "" +
+                "{" +
+                "  \"properties\": { " +
+                "    \"keyword_column\":   { \"type\": \"keyword\" }," +
+                "    \"text_column\":      { \"type\": \"text\" }" +
+                "  }" +
+                "}";
```
Comment (martint) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:1552]: Use a text block for better readability:

```suggestion
        String mappings = """
                {
                  "properties": {
                    "keyword_column":   { "type": "keyword" },
                    "text_column":      { "type": "text" }
                  }
                }
                """;                
```

**Review by martint — COMMENTED** [review]
The OpenSearch connector probably has the same issue. Can you apply the fix there, too?

**Comment by bvolpato** [comment]
> The OpenSearch connector probably has the same issue. Can you apply the fix there, too?

Good call, yes, just reproduced there with the same set of tests. Pushed the same change there.

I also applied the text blocks change/suggestions too, agree that it looks much cleaner. For context, I followed the same structure from the surrounding tests -- so they could likely get a similar refactoring.

**Comment by bvolpato** [comment]
Lastly, I've submitted the CLA, but I guess it might take a couple of days to hear back.

**Comment by pettyjamesm** [comment]
@martint - looks like this is just pending CLA processing before it can merge.

**Comment by martint** [comment]
@cla-bot check

**Review by hashhar — APPROVED** [review]

**Pull request merged**